### ISSUE TITLE
Add the Content of the error response inside the SparkException object

### ISSUE THREAD

**Issue by Amebus** [darrenparkinson/SparkDotNet#25]
When Webex apis go in error they return a full description of the error inside the response content

E.g. a call to https://webexapis.com/v1/people (done with postman) returns a 403 Forbidden with the following content if you are not an admin:

```json
{
    "message": "Only admin is allowed to use this resource.",
    "errors": [
        {
            "description": "Only admin is allowed to use this resource."
        }
    ],
    "trackingId": "ROUTER_5F11AD84-1BF8-01BB-4B57-AC12DA2C4B57"
}
```

But in the SparkException there is no trace of the content of the answer (or at least i wasn't able to find it).
And looking in the code I see that the message is set to the error code:

```csharp
throw new SparkException((int)response.StatusCode, $"{(int)response.StatusCode} ({response.ReasonPhrase})", response.Headers)
```

It would be nice and useful (at least for debugging and logging purposes) to have the response content also in the SparkException object.

As a side note: it seems that the structure of the error content is always the same:

```csharp
public class SparkErrorContent
{
    public string Message { get; set; }
    public List<SparkErrorMessage> Errors { get; set; }
    public string TrackingId { get; set; }
}
public class SparkErrorMessage
{
    public string Description { get; set; }
}
```

**Comment by darrenparkinson** [comment]
Hi. Apologies for the delay.  So would you essentially want an additional field on SparkException? So in addition to `StatusCode` and `Headers` something like `Content` of type `SparkErrorContent` -- something along those lines?

**Comment by Amebus** [comment]
Hi, yes, the idea is to provide an additional field in the SparkException which contains the content of the response of the Webex api.

In this way I can see the error that triggered the exception and act accordingly, for example logging the error or reporting the error to an administrator of my system

Maybe the message of the SparkException can be exactly the Message of the SparkErrorContent, do you think is a good idea?

**Comment by tcs-devel** [comment]
As we also require the possibility of accessing the error information, we created a fork of the code and implemented support for it based on the idea above.

We decided to use `SparkErrorContent.Message` as message of the `SparkException` with a fallback to the `ReasonPhrase` of the response.

PR #30 contains the changes, would be nice if the change could somehow be integrated into your code.

**Comment by darrenparkinson** [comment]
Thanks for the PR.  Is this a breaking change?  Does it require any update to the README?  Thanks.

**Comment by tcs-devel** [comment]
The only change in "behavior" we made is, that the message of the `SparkException` does not contain the ReasonPhrase anymore (or better: only as a fallback if the message of the error itself is null).

There should be no other change in behavior inside the library, `SparkException`s are still thrown for all requests if no success response was received. Unused constructors of `SparkException` have been removed.
